src/Paranoia/Payment/Adapter/Gvp.php

```diff
@@ -390,23 +516,46 @@ protected function parseResponse($rawResponse, $transactionType)
             $response->setResponseMessage('Success');
             $response->setOrderId((string)$xml->Order->OrderID);
             $response->setTransactionId((string)$xml->Transaction->RetrefNum);
+            $response->setAuthCode((string)$xml->Transaction->AuthCode);
         }
         $event = $response->isSuccess() ? self::EVENT_ON_TRANSACTION_SUCCESSFUL : self::EVENT_ON_TRANSACTION_FAILED;
         $this->getDispatcher()->dispatch($event, new PaymentEventArg(null, $response, $transactionType));
         return $response;
     }
 
+    protected function check3DHashIntegrity($payload) {
+        $params    = explode(':', $payload['hashparams']);
+        $secureKey = $this->configuration->getSecureKey();
+        $hash      = '';
+
+        foreach($params as $param) {
+            if (!empty($param))
+                $hash .= is_null($payload[$param]) ? '' : $payload[$param];
+        }
+
+        if($this->hashBase64($hash . $secureKey) == $payload['hash'])
+            return true;
+
+        return false;
+    }
+
+    protected function parseBank3DResponse($rawResponse)
+    {
+        $response = new PaymentResponse();
+        $response->setOrderId($rawResponse['orderid']);
+        $response->setTransactionId($rawResponse['orderid']);
+        $response->setMdStatus($rawResponse['mdstatus']);
+        $response->setResponseMessage($rawResponse['mderrormessage']);
+        return $response;
+    }
+
     /**
      * {@inheritdoc}
      * @see Paranoia\Payment\Adapter\AdapterAbstract::formatAmount()
      */
     protected function formatAmount($amount, $reverse = false)
     {
-        if (!$reverse) {
-            return number_format($amount, 2, '', '');
-        } else {
-            return (float)sprintf('%s.%s', substr($amount, 0, -2), substr($amount, -2));
-        }
+        return str_replace(".", "", number_format($amount, 2, '.', ''));
```

Comment:
Yaptığın değişiklik $reverse parametresini etkisiz hale getirmiş. Aynı fonksiyon hem adaptöre verirken, hemde adaptördeki değeri geri alırken kullanılıyor. (bunlar iki ayrı fonksiyon olabilir. Geçmiş uyumluluğunu bozacağı için bir sonraki major sürümde değiştirebiliriz.)